src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AccessController.cs

```diff
@@ -490,7 +501,17 @@ private async Task<IActionResult> ExchangePasswordGrantType(OpenIddictRequest re
                 {
                     [OpenIddictServerAspNetCoreConstants.Properties.Error] = Errors.UnsupportedGrantType,
                     [OpenIddictServerAspNetCoreConstants.Properties.ErrorDescription] =
-                        "The resource owner password credentials grant is not supported."
+                        S["The resource owner password credentials grant is not supported."]
+                }), OpenIddictServerAspNetCoreDefaults.AuthenticationScheme);
+            }
+
+            var disableLocalLogin = (await _siteService.GetSiteSettingsAsync()).As<LoginSettings>().DisableLocalLogin;
```

Comment:
We should probably have that logic directly in `UserService.AuthenticateAsync()`.

Reply:
Should we? What if we want to authenticate using other services while the local login in disabled?

Reply:
This method is only used for local logins (it's backed by Identity's `CheckPasswordSignInAsync()`, which is a password-only thing anyway).